### PR TITLE
Tdigest

### PR DESCRIPTION
**Pull request by DuanKuanJun**

### CODE REVIEW

**Comment by hjxilinx** [comment]
where is the test script and unit test?

File: src/query/inc/tdigest.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#ifndef M_PI
+#define M_PI        3.14159265358979323846264338327950288   /* pi             */
+#endif
+
+#define COMPRESSION 400
+#define GET_CENTROID(compression)  (ceil(compression * M_PI / 2) + 1)
+#define GET_THRESHOLD(compression) (7.5 + 0.37 * compression - 2e-4 * pow(compression, 2))
+#define TDIGEST_SIZE(compression)  (sizeof(TDigest) + sizeof(Centroid)*GET_CENTROID(compression) + sizeof(Point)*GET_THRESHOLD(compression))
+
+typedef struct Centroid {
```
Comment (hjxilinx) [inline src/query/inc/tdigest.h]: The name of the struct should begin with the capital letter of "S".

File: src/query/src/tdigest.c [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+ * It is based on the MergingDigest implementation found at:
+ *   https://github.com/tdunning/t-digest/blob/master/src/main/java/com/tdunning/math/stats/MergingDigest.java
+ *
+ * Copyright (c) 2016, Usman Masood <usmanm at fastmail dot fm>
+ */
+
+#include <ctype.h>
+#include <float.h>
+#include <math.h>
+#include <stdint.h>
```
Comment (hjxilinx) [inline src/query/src/tdigest.c]: Here those includes should be replaced by `#include  "os.h"`

Comment (DuanKuanJun) [reply src/query/src/tdigest.c]: already fixed

**Pull request merged**